### ISSUE TITLE
Add indexes for search performance

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#6301]
From https://forum.pkp.sfu.ca/t/ojs-3-2-database-takes-to-long/63080/15:
> We’ve been seeing that our database takes too long to make a response to some querys like this one...

It appears that search queries are imposing a heavy load, and two indexes in particular are missing:
- submissions.current_publication_id
- submission_search_objects.submission_id

**Comment by asmecher** [comment]
Users can create the missing indexes using the following SQL statements.
```
CREATE INDEX submission_search_object_submission ON submission_search_objects (submission_id);
CREATE INDEX submissions_publication_id ON submissions (current_publication_id);
```

**Comment by NateWr** [comment]
When testing migrations in postgres for #6057 I ran into errors related to the migration for these indexes:

```
ERROR: Upgrade failed: DB: SQLSTATE[42P07]: Duplicate table: 7 ERROR:  relation "submissions_publication_id" already exists (SQL: create index "submissions_publication_id" on "submissions" ("submission_id"))
```

**Comment by glaucomunsberg** [comment]
Is this still an issue in 3.3.0.8 @asmecher, to fix the slow load in my journal. To fix it I had to run the query below
```sql
CREATE INDEX submission_search_object_submission ON submission_search_objects (submission_id);
CREATE INDEX submissions_publication_id ON submissions (current_publication_id);
```

**Comment by asmecher** [comment]
@glaucomunsberg, I can confirm that the statements to add those indexes are included with 3.3.0-x, both for new installations ([1](https://github.com/pkp/pkp-lib/blob/655a531801aac74b387881454c3d046e020e3488/classes/migration/SubmissionsMigration.inc.php#L167) [2](https://github.com/pkp/pkp-lib/blob/655a531801aac74b387881454c3d046e020e3488/classes/migration/SubmissionsMigration.inc.php#L43)) and [for upgrades](https://github.com/pkp/pkp-lib/blob/655a531801aac74b387881454c3d046e020e3488/classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php#L71..L83). I wonder if there was some quirk in your upgrade process that resulted in them not being added?

**Comment by jonasraoni** [comment]
@glaucomunsberg there are some tools out there to compare databases, I'd advise you to issue a clean installation, using the same version that you have, and check if at least the structure match with your database.
I consider this is a good approach to start, then you might check if the records created by the system are intact and so on.

**Comment by diegomejia071** [comment]
Hi @asmecher ,

I can create these indexes for my OJS 3.2.1.4 installation, since its loading is very slow when showing the back issues, when published and unpublished and when performing tasks in the shipments that in the version that had before updating they loaded normally.

**Comment by asmecher** [comment]
@diegomejia071, I'm not sure I'm following your last comment. Were you able to add the indexes for your 3.2.1-4 installation? Did they improve the behaviour?

**Comment by asmecher** [comment]
@diegomejia071, I think that `Duplicate key name` message simply means that the indexes already exist.

**Comment by diegomejia071** [comment]
Hi @asmecher,

I understand that the indexes are duplicated, but then why does it keep taking so long. As I mention it here.

https://forum.pkp.sfu.ca/t/ojs-3-2-1-4-published-and-unpublished-stay-loading-too-long/70737

**Comment by asmecher** [comment]
That is unrelated to this issue, I think.

**Comment by diegomejia071** [comment]
Hi @asmecher,

And I have several loading problems of different aspects in the ojs, such as publishing and unpublishing since it takes a long time, to review the back issues is the same, so I do not know what can cause this.

**Comment by asmecher** [comment]
Stay tuned and someone will follow up to your forum post.

**Comment by alshammaa** [comment]
Hi @asmecher,
I noticed that the added index for submissions.current_publication_id in the [upgrade code](https://github.com/pkp/pkp-lib/blob/655a531801aac74b387881454c3d046e020e3488/classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php#L71..L83) is made on submission_id column not current_publication_id column. Is this right? or it should be chnaged?

**Comment by asmecher** [comment]
@alshammaa, you're correct, it should be `current_publication_id` and not `submission_id` -- but our policy is not to introduce database changes in builds, so this won't be fixed until 3.4.0 (unless you correct it manually, which will not have side-effects for the upgrade). The commit that fixes it for 3.4 is: 0223fc3a6d7f4526c326d38cd1dc338380b5b152
